### PR TITLE
Refactor: change variables naming under context of Agent Framework

### LINKED ISSUE #96: [Refactor]Standardize variables' name under context of Agent framework
- [x] trace_id -> interactionId
- [x] session_id -> conversationId

### PR DESCRIPTION
**Pull request by SuZhou-Joe**
### Description
1. Change variable names from sessionId to conversationId.
2. Change file names from session-related to conversation-related.

### Issues Resolved
#96 

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test.
- [x] New functionality has user manual doc added.
- [x] Commits are signed per the DCO using --signoff.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by SuZhou-Joe** [comment]
Did a regression test on:

- [ ] Loading conversations and conversation
- [ ] Showing how was it generated
- [ ] Change conversation title
- [ ] Delete conversation
- [ ] Search conversation
- [ ] Regenerate interaction

File: server/routes/get_session.test.ts
Comment (wanglam) [inline server/routes/get_session.test.ts:1]: Shall we rename this get_session test file to get_conversation?

File: public/hooks/__tests__/use_sessions.test.ts
Comment (wanglam) [inline public/hooks/__tests__/use_sessions.test.ts:1]: Shall we update filename?

File: server/routes/get_sessions.test.ts
Comment (wanglam) [inline server/routes/get_sessions.test.ts:1]: Shall we update filename?

File: server/routes/get_session.test.ts
Comment (SuZhou-Joe) [reply server/routes/get_session.test.ts:1]: Sure, done.

File: server/routes/get_sessions.test.ts
Comment (SuZhou-Joe) [reply server/routes/get_sessions.test.ts:1]: Sure, done.

**Review by wanglam — APPROVED** [review]

**Pull request merged**